### PR TITLE
LCA 4.14/4.16/main branch update

### PR DESCRIPTION
**Pull request by jc-rh**

### CODE REVIEW

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-main-ibu-e2e-flow

File: ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  cluster: build02
+  optional: true
+  skip_if_only_changed: ^docs/|LICENSE|OWNERS|\.md$|^hack$
+  steps:
+    cluster_profile: aws
+    dependencies:
+      LCA_PULL_REF: pipeline:lifecycle-agent-operator
+    env:
+      OCP_BASE_VERSION: "4.14"
+      OCP_IMAGE_SOURCE: ci
```
Comment (jeff-roche) [inline ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml:96]: I would make this release to test against the latest released image

Comment (jeff-roche) [reply ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml:96]: The e2e flow should also probably be added to the 4.15 branch

Comment (jc-rh) [reply ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml:96]: Apologize if I asked about this before, what exactly is this version?

Comment (jeff-roche) [reply ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml:96]: `OCP_BASE_VERSION` is the version of OCP that will be used for the seed and target clusters
`OCP_IMAGE_SOURCE` determines where those images are pulled from (`ci` is images built in CI, `release` will use officially publically released images of OCP. I'm of the opinion that for release branches, we should be testing against released OCP images

Comment (jeff-roche) [reply ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml:96]: Also, this comment was merely stating that the config file for release-4.15 does not have this e2e workflow in it

Comment (jc-rh) [reply ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml]: Updated

**Comment by jc-rh** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.16-image-based-upgrade-e2e-parallel-conformance

**Comment by jc-rh** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.16-image-based-upgrade-e2e-serial-conformance

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-4.16-images

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-4.16-integration

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-4.16-markdownlint

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-4.16-security

**Comment by jc-rh** [comment]
Periodic conformance tests are expected to fail.
Will address 4.14/4.15 e2e test in a subsequent PR.

File: ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-main.yaml
Comment (nishant-parekh) [inline ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-main.yaml:1]: Does the `OCP_BASE_VERSION: "4.16"` on line 95 need to be updated as well?

File: ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.15.yaml [hunk 2]
```diff
@@ -83,6 +83,28 @@ tests:
...
+  optional: true
+  skip_if_only_changed: ^docs/|LICENSE|OWNERS|\.md$|^hack$
+  steps:
+    cluster_profile: aws
+    dependencies:
+      LCA_PULL_REF: pipeline:lifecycle-agent-operator
+    env:
+      OCP_BASE_VERSION: "4.15"
+      OCP_IMAGE_SOURCE: release
+      SEED_IMAGE_TAG_FORMAT: presubmit
```
Comment (jeff-roche) [inline ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.15.yaml:97]: this should be `release`

File: ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.16.yaml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  optional: true
+  skip_if_only_changed: ^docs/|LICENSE|OWNERS|\.md$|^hack$
+  steps:
+    cluster_profile: aws
+    dependencies:
+      LCA_PULL_REF: pipeline:lifecycle-agent-operator
+    env:
+      OCP_BASE_VERSION: "4.16"
+      OCP_IMAGE_SOURCE: ci
+      SEED_IMAGE_TAG_FORMAT: presubmit
```
Comment (jeff-roche) [inline ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.16.yaml:97]: this should be `release`

File: ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  optional: true
+  skip_if_only_changed: ^docs/|LICENSE|OWNERS|\.md$|^hack$
+  steps:
+    cluster_profile: aws
+    dependencies:
+      LCA_PULL_REF: pipeline:lifecycle-agent-operator
+    env:
+      OCP_BASE_VERSION: "4.14"
+      OCP_IMAGE_SOURCE: release
+      SEED_IMAGE_TAG_FORMAT: presubmit
```
Comment (jeff-roche) [inline ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.14.yaml:97]: this should be `release`

**Review by jeff-roche — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-main.yaml
Comment (jc-rh) [reply ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-main.yaml:1]: 4.17 images are not available yet.

**Comment by jc-rh** [comment]
/pj-rehearse ci/rehearse/openshift-kni/lifecycle-agent/main/security

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-4.16-images

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-4.16-security

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-release-main-security

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-main-integration

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-main-ibu-e2e-flow

**Comment by jc-rh** [comment]
/pj-rehearse pull-ci-openshift-kni-lifecycle-agent-main-security

**Comment by jc-rh** [comment]
@nishant-parekh This PR is ready. Periodic conformance tests are expected to fail.

**Pull request merged**